src/PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/CovariantDerivOfExtrinsicCurvature.cpp

```diff
@@ -0,0 +1,136 @@
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal_vector,
+    const tnsr::Ijj<DataType, SpatialDim, Frame>&
+        spatial_christoffel_second_kind,
+    const tnsr::ii<DataType, SpatialDim, Frame>& extrinsic_curvature,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& d_pi,
+    const tnsr::ijaa<DataType, SpatialDim, Frame>& d_phi) noexcept {
+  tnsr::ijj<DataType, SpatialDim, Frame> d_extrinsic_curvature{};
+  GeneralizedHarmonic::covariant_deriv_of_extrinsic_curvature<SpatialDim, Frame,
+                                                              DataType>(
+      make_not_null(&d_extrinsic_curvature), spacetime_unit_normal_vector,
+      spatial_christoffel_second_kind, extrinsic_curvature,
+      inverse_spacetime_metric, phi, d_pi, d_phi);
+  return d_extrinsic_curvature;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void covariant_deriv_of_extrinsic_curvature(
+    const gsl::not_null<tnsr::ijj<DataType, SpatialDim, Frame>*>
+        d_extrinsic_curvature,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal_vector,
+    const tnsr::Ijj<DataType, SpatialDim, Frame>&
+        spatial_christoffel_second_kind,
+    const tnsr::ii<DataType, SpatialDim, Frame>& extrinsic_curvature,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& d_pi,
+    const tnsr::ijaa<DataType, SpatialDim, Frame>& d_phi) noexcept {
+  if (UNLIKELY(get_size(get<0, 0, 0>(*d_extrinsic_curvature)) !=
+               get_size(get<0>(spacetime_unit_normal_vector)))) {
+    *d_extrinsic_curvature = tnsr::ijj<DataType, SpatialDim, Frame>(
+        get_size(get<0>(spacetime_unit_normal_vector)));
+  }
+
+  // Ordinary derivative first
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t j = i; j < SpatialDim; ++j) {
+      for (size_t k = 0; k < SpatialDim; ++k) {
+        d_extrinsic_curvature->get(k, i, j) = 0.5 * d_pi.get(k, i + 1, j + 1);
+        for (size_t a = 0; a <= SpatialDim; ++a) {
+          d_extrinsic_curvature->get(k, i, j) +=
+              0.5 * (d_phi.get(k, i, j + 1, a) + d_phi.get(k, j, i + 1, a)) *
+              spacetime_unit_normal_vector.get(a);
+          for (size_t b = 0; b <= SpatialDim; ++b) {
+            for (size_t c = 0; c <= SpatialDim; ++c) {
+              d_extrinsic_curvature->get(k, i, j) -=
+                  0.5 * (phi.get(i, j + 1, a) + phi.get(j, i + 1, a)) *
+                  spacetime_unit_normal_vector.get(b) *
+                  (inverse_spacetime_metric.get(c, a) +
+                   0.5 * spacetime_unit_normal_vector.get(c) *
+                       spacetime_unit_normal_vector.get(a)) *
+                  phi.get(k, c, b);
```

Comment:
I verified that this is the correct partial derivative of $K_{ij}$, except that I don't know where the identity for $\partial_k t^a$ comes from

Reply:
Yup, updated the help text to include the derivation of what is coded up here.